src/app/pages/AccountDetailsPage/hook.ts

```diff
@@ -0,0 +1,73 @@
+import { useRequiredScopeParam } from '../../hooks/useScopeParam'
+import {
+  EvmTokenType,
+  Layer,
+  RuntimeEvmBalance,
+  useGetRuntimeAccountsAddress,
+  useGetRuntimeTransactions,
+} from '../../../oasis-indexer/api'
+import { useLoaderData } from 'react-router-dom'
+import { AppErrors } from '../../../types/errors'
+import { useSearchParamsPagination } from '../../components/Table/useSearchParamsPagination'
+import { NUMBER_OF_ITEMS_ON_SEPARATE_PAGE } from '../../config'
+
+export const useAccount = () => {
+  const { network, layer } = useRequiredScopeParam()
+  if (layer === Layer.consensus) {
+    // There can be no ERC-20 or ERC-721 tokens on consensus
+    throw AppErrors.UnsupportedLayer
+  }
+  const address = useLoaderData() as string
```

Comment:
I think components referenced from routes (page components) should be the ones that obtain parameters from the URL, not API hooks

Reply:
Fine. Will refactor.

Reply:
Done.